specifyweb/stored_queries/tests.py

```diff
@@ -476,6 +476,29 @@ def test_relationships_in_switch_fields(self):
             query = query.query.add_columns(expr)
             self.assertCountEqual(list(query), [('text 1 value for this accession',), (' text 2 value for this accession role2',)])
 
+    def test_group_by_with_aggregators(self):
+      with models.session_context() as session:
+        prep_col = func.max(getattr(models.Preparation, models.Preparation._id))
+        
+        query = (
+            select([prep_col])
+            .select_from(models.Preparation)
+            .where(models.Preparation.CollectionObjectID == models.CollectionObject.collectionObjectId)
+            .correlate(models.CollectionObject)
+            .as_scalar()
+        )
+        
+        new_query = (
+            session.query(models.CollectionObject._id)
+            .select_from(models.CollectionObject)
+            .filter(models.CollectionObject.CollectionID == 1)
+            .add_column(query)
+            .group_by(query)
+        )
+        
+        str_query = str(new_query.statement.compile(compile_kwargs={"literal_binds": True}))
+        rr = list(new_query)
```

Comment:
Hmm I'd say add a unit test that tests if sql statement generated is correct or not. See some other unit tests above that generate sql for expressions and compare it.

Reply:
Ya, we could do that, I was just using the test to check that it the query generated is valid.  So as long as no exception is created, the tests passes.